### PR TITLE
pilot: set `istio.alpn_override: false` in subset clusters

### PR DESCRIPTION
**Pull request by jewertow**
**Please provide a description of this PR:**

This is a follow-up to #44918.

That PR didn't cover a case where TLS policy is configured for a subset, like this:
```
apiVersion: networking.istio.io/v1beta1
kind: DestinationRule
metadata:
  name: originate-simple-tls
spec:
  host: app.default.svc.cluster.local
  subsets:
  - name: v1
    trafficPolicy:
      tls:
        mode: SIMPLE
```

### CODE REVIEW

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -395,9 +395,13 @@ func AddSubsetToMetadata(md *core.Metadata, subset string) {
 	}
 }
 
-// AddALPNOverrideToMetadata adds whether the the ALPN prefix should be added to the header
-// to the given core.Metadata struct, if metadata is not initialized, build a new metadata.
-func AddALPNOverrideToMetadata(metadata *core.Metadata, alpnOverride bool) *core.Metadata {
```
Comment (jewertow) [inline pilot/pkg/networking/util/util.go:400]: JFYI: alpn_override is always set to "false", and this is very unlikely that we will need to set "true", so I decided to refactor this function to simplify cluster builder.

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 2]
```diff
@@ -218,6 +218,7 @@ func (cb *ClusterBuilder) buildSubsetCluster(opts buildClusterOpts,
 	// basis in buildCluster, so we can just insert without a copy.
 	subsetCluster.cluster.Metadata = util.AddConfigInfoMetadata(subsetCluster.cluster.Metadata, destRule.Meta)
 	util.AddSubsetToMetadata(subsetCluster.cluster.Metadata, subset.Name)
+	subsetCluster.cluster.Metadata = util.AddALPNOverrideToMetadata(subsetCluster.cluster.Metadata, opts.policy.GetTls().GetMode())
```
Comment (jewertow) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:221]: As you can see, `util.AddSubsetToMetadata` does not return metadata, it does not initialize metadata if it's nil. Do you think that we should follow that semantics?

**Review by howardjohn — APPROVED** [review]

**Pull request merged**